clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1001,9 +1001,14 @@ protected void close(Timer timer) {
     }
 
     /**
+     * Leave the group. This method also sends LeaveGroupRequest and log {@code leaveReason} if this is dynamic members
+     * or unknown coordinator or state is not UNJOINED or this generation has a valid member id.
+     *
+     * @param leaveReason the reason to leave the group for logging
+     * @param shouldWarn should log as WARN level or INFO
      * @throws KafkaException if the rebalance callback throws exception
      */
-    public synchronized RequestFuture<Void> maybeLeaveGroup(String leaveReason) {
+    public synchronized RequestFuture<Void> maybeLeaveGroup(String leaveReason, boolean shouldWarn) throws KafkaException {
```

Comment:
Most cases pass `false` so an override method  `maybeLeaveGroup(String)` can reduce the code changes. Also, `maybeLeaveGroup(String, boolean)` can be declared as a protected method. WDYT?

Reply:
Cool! And, it cannot change to `protected` method since we used this method in `KafkaConsumer`, which is in different package.